load.php

```diff
@@ -462,6 +462,27 @@ function perflab_run_module_activation_deactivation( $old_value, $value ) {
 			if ( ! empty( $module_settings['enabled'] ) && ( empty( $old_value[ $module ] ) || empty( $old_value[ $module ]['enabled'] ) ) ) {
 				perflab_activate_module( PERFLAB_PLUGIN_DIR_PATH . 'modules/' . $module );
 			}
+
+			// Check whether it is necessary to display an admin pointer to prompt the user to migrate.
+			$plugin_slug        = basename( $module );
+			$standalone_plugins = perflab_get_standalone_plugins();
+
+			// Plugin slug should be in standalone plugins list.
+			if ( ! in_array( $plugin_slug, $standalone_plugins, true ) ) {
+				continue;
+			}
+
+			$current_user = get_current_user_id();
```

Comment:
This will only work for the current user, but modules are active for the site. In other words, it's not working for all relevant users. We would need to use `WP_User_Query` for that, with a meta query for the `dismissed_wp_pointers` key (and possibly even with a `LIKE` for the `perflab-module-migration-pointer` value. That way we would unset the value for all relevant users.

Realistically, this is only a problem for sites that have more than 1 admin, which speaking at scale is not a ton. So I think we could also make this a future enhancement PR to unblock this. But I would suggest we implement it sooner or later.

Reply:
While I appreciate the proposed solution, I'd like to highlight a potential challenge we might encounter. Consider a scenario where a site has two administrators, one catering to the non-technical team and the other to the technical team.

If we implement suggested changes. The behaviour with the migration pointer is as follows:

- Upon plugin release, the migration pointer becomes visible on the site.
- When the non-technical team logs in, they see the migration pointer.
- If they dismiss the pointer, it won't appear for them again.
- However, the technical team, when logging in, never encounters this migration pointer.

The concern with implementing the suggested solution is that it would remove the pointer for all users across the site, affecting both the non-technical and technical teams. This could potentially lead to the technical team missing out on crucial information conveyed by the pointer.

Looking forward to further discussion on this matter.

Reply:
> If we implement suggested changes. The behaviour with the migration pointer is as follows:
> 
> * Upon plugin release, the migration pointer becomes visible on the site.
> * When the non-technical team logs in, they see the migration pointer.
> * If they dismiss the pointer, it won't appear for them again.
> * However, the technical team, when logging in, never encounters this migration pointer.

That wouldn't be the case, maybe there was a misunderstanding or I didn't explain it well. The pointer wouldn't be dismissed for every user, it would only be "un-dismissed" for every user.

In other words:
* User A dismisses the pointer.
* User B still sees the pointer.
* User B performs the migration.
* Later, user B for some reason deactivates a standalone plugin and reactivates the corresponding module again.
* Now there's the difference:
    * Based on the current PR code, only user B would see the pointer again, but the problem applies to the entire site, so I think user A should see the pointer again as well.
    * In my proposal, both user B and user A would see the pointer again as a result of the change from user B, so user A would be aware of the problem and then could fix it (e.g. if user B goes on holidays).

Reply:
Thanks @felixarntz for explaining. Opened followup PR https://github.com/WordPress/performance/pull/917.